src/PAModel/MsAppTest.cs

```diff
@@ -14,6 +16,126 @@ namespace Microsoft.PowerPlatform.Formulas.Tools
+                var doc2New = CanvasMerger.Merge(doc2, doc1, doc1);
+                var ok2 = HasNoDeltas(doc2, doc2New);
+
+                return ok1 && ok2;
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine(e.Message);
+                return false;
+            }
+        }
+
+        public static bool DiffStressTest(string pathToMsApp)
+        {            
+            (CanvasDocument doc1, var errors) = CanvasDocument.LoadFromMsapp(pathToMsApp);
+            errors.ThrowOnErrors();
+
+            return HasNoDeltas(doc1, doc1);
+        }
+
+        // Verify there are no deltas (detected via smart merge) between doc1 and doc2
+        private static bool HasNoDeltas(CanvasDocument doc1, CanvasDocument doc2)
+        {
+            var ourDeltas = Diff.ComputeDelta(doc1, doc1);
+
+            // ThemeDelta always added
+            ourDeltas = ourDeltas.Where(x => x.GetType() != typeof(ThemeChange)).ToArray();
+
+            if (ourDeltas.Any())
+            {
+                foreach (var diff in ourDeltas)
+                {
+                    Console.WriteLine($"  {diff.GetType().Name}");
+                }
+                // Error! app shouldn't have any diffs with itself.
+                return false;
+            }
+
+
+            // Save and verify checksums.
+            using (var temp1 = new TempFile())
+            using (var temp2 = new TempFile())
+            {
+                doc1.SaveToMsApp(temp1.FullPath);
+                doc2.SaveToMsApp(temp2.FullPath);
+
+                var doc1NoEntropy = RemoveEntropy(temp1.FullPath);
+                var doc2NoEntropy = RemoveEntropy(temp2.FullPath);
+
+                Compare(doc1NoEntropy, doc2NoEntropy, Console.Out);
+            }
+
+            return true;
+        }
+
+        // Unpack, delete the entropy dirs, repack. 
+        public static CanvasDocument RemoveEntropy(string pathToMsApp)
+        {
+            using (var temp1 = new TempDir())
+            {
+                (CanvasDocument doc1, var errors) = CanvasDocument.LoadFromMsapp(pathToMsApp);
+                errors.ThrowOnErrors();
+
+                doc1.SaveToSources(temp1.Dir);
+
+                var entropyDir = Path.Combine(temp1.Dir, "Entropy");
+                if (!Directory.Exists(entropyDir))
+                {
+                    throw new Exception($"Missing entropy dir: " + entropyDir);
+                }
+
+                Directory.Delete(entropyDir, recursive: true);
+
+                (var doc2, var errors2) = CanvasDocument.LoadFromSources(temp1.Dir);
+                errors.ThrowOnErrors();
+
+                
+#if false
+                // $$$ We get different checksums when deleting in-memory vs. files.
+                // In some cases, files are just rearranged (\images),
+                // but in other cases, look different. 
+
+                // Reset Entropy on in-memory model
+                doc1._entropy = new Entropy();
```

Comment:
This constantly fails too. The expectation (perhaps wrongly?) is that clearing the entropy in memory would have same affect as deleting entropy on disk. 
In reality, lots of unexpected diffs.